README.md

```diff
@@ -1213,46 +1213,48 @@ func TestFn(t *testing.T) {
 
-But first and foremost. What drives developers to use the empty `interface{}`? Well, as I said in the previously, the way that Go determines whether a concrete type implements an interface, is by checking whether it implements the methods of a specific interface. So what happens, if our interface implement no methods at all?
+As mentioned in a previous section, Go determines whether a concrete type implements a particular interface by checking whether the type implements the <em>methods</em> of that interface. So what happens if our interface declares no methods, as is the case with the empty interface?
 
 ```go
 type EmptyInterface interface {}
 ```
 
-The above being equivalent to the built-in type `interface{}`. The result of this interface type is that **any** type is accepted. Meaning, that we can write functions in which any type is accepted. This is super useful for certain kind of functions, such as when creating a printer function. This is how it's possible to give any type to the `Println` function from the `fmt` package:
+The above is equivalent to the built-in type `interface{}`. A natural consequence of this is that we can write generic functions&mdash;ones that accept any type as arguments. This is extremely useful for certain kinds of functions, such as print helpers. Interestingly, this is actually what makes it possible to pass in any type to the `Println` function from the `fmt` package:
 
 ```go
 func Println(v ...interface{}) {
     ...
 }
 ```
 
-In this case, we aren't only accepting a single `interface{}` but rather, a slice of types. These types can be of any type and can even be of different types, as long as they implement the empty `interface{}`, which we are certain that any type will. This is a super common pattern when handling string conversation (both from and to string). The reason being, this is the only way in Go to implement generic methods. Good examples of this, come from the `json` standard library package:
+In this case, `Println` isn't just accepting a single `interface{}`; rather, the function accepts a <em>slice</em> of types. These can be <em>any</em> types, not necessarily all the same. The only requirement is that all of the types implement the empty `interface{}`, which we are certain that any type will. This is a very common pattern when handling string conversation (both from and to a string). Good examples of this come from the `json` standard library package:
```

Comment:
One of those times where I feel that a sentence should be a little different, but not sure, if below is an improvement?

```suggestion
In this case, `Println` isn't just accepting a single `interface{}`; rather, the function accepts a <em>slice</em> of types. The only requirement is that all of the types implement the empty `interface{}`, which we are certain that any type will. This enables `Println` to accept a slice of <em>any</em> type, whose elements aren't restricted to being uniform. This is a very common pattern when handling string conversation (both from and to a string). Good examples of this come from the `json` standard library package:
```

Reply:
Hmm, I think we end up repeating `the function accepts a <em>slice</em> of types` when we say `This enables `Println` to accept a slice of <em>any</em> type, whose elements aren't restricted to being uniform.`

I do agree that my edit is a bit wordy, though. What do you think of this alternative?

```
According to the signature, `Println` accepts a single `interface{}`. In reality, this means it accepts a <em>slice</em> of any type, whose elements need not be uniform. This is a very common pattern...
```

Reply:
Hmm... how about this instead then?

```suggestion
In this case, `Println` isn't just accepting a single `interface{}`; rather, the function accepts a <em>slice</em> of types that implement the empty `interface{}`. As there are no methods associated with the empty `interface{}` <em>all</em> types are accepted, even making it possible to feed `Println` with a slice of different types. This is a very common pattern when handling string conversation (both from and to a string). Good examples of this come from the `json` standard library package:
```

Reply:
Yep, that works! Just be sure to add a comma after ``interface{}`` :)

Reply:
I'm guessing you mean before the <em>all</em>?

```suggestion
In this case, `Println` isn't just accepting a single `interface{}`; rather, the function accepts a <em>slice</em> of types that implement the empty `interface{}`. As there are no methods associated with the empty `interface{}`, <em>all</em> types are accepted, even making it possible to feed `Println` with a slice of different types. This is a very common pattern when handling string conversation (both from and to a string). Good examples of this come from the `json` standard library package:
```

Reply:
Oh, yeah, there :)